### DISCUSSION TITLE
Very weak signal between 2 Sx1262

### DISCUSSION THREAD

**Discussion by Baptou88** [jgromes/RadioLib#653] [Q&A]
Good morning all,

I currently have a few boards from Heltec which are equipped for version 2 with an ESP32 and an SX1276 and for version 3 with an ESP32S3 and an SX1262.

I am able to make them all speak together, however I am very surprised at the low RSSI. I am currently developing on my desk with 2 SX1262 and the RSSI is -105dbm (the boards are only 1m apart, is that normal?)

Strangely, I communicate with 1 SX1276 105m away and the signal is -95dbm

Here is a preview of my code, which can be seen  [here](https://github.com/Baptou88/HydroBabiat/blob/master/lib/HydroBabiatLib/src/LoRa.cpp#L14):
```cpp
int LoRaClass::begin(){

#ifdef ARDUINO_HELTEC_WIFI_LORA_32_V2
    radio = new Module(SS, 26, RST_LoRa, 33);
#elif defined(ARDUINO_HELTEC_WIFI_LORA_32_V3)
    radio = new Module(LoRa_SS, 14, LoRa_RST, 13);
#else
    #error UnImplemented
#endif

    SPI.begin(LoRa_SCK,LoRa_MISO,LoRa_MOSI,LoRa_SS);
    
  
    radio.setSyncWord(0x12);
    delay(10);
    radio.setBandwidth(125);
    delay(10);
    radio.setFrequency(868);
    delay(10);
    radio.setCodingRate(7);
    delay(10);

  
    
#endif

    #ifdef ARDUINO_HELTEC_WIFI_LORA_32_V2
        radio.setDio0Action(LoRaClass::setFlag);
    #elif defined(ARDUINO_HELTEC_WIFI_LORA_32_V3)
        radio.setDio1Action(LoRaClass::setFlag);
    #else
        #error UnImplemented
    #endif

    return radio.begin(868.0,125,9,7,18);
```

what can i do ?
Respectfully Baptiste

**Comment by jgromes** [comment]
-105 dBm is low, but the cause of that can be loads of things, from mismatched antenna to insufficient power source, so unless you show some more details about your setup, there's not much that can be guessed.

Also you are initializing the module in a very strange way.

1. Why are you explicitly calling SPI.begin?
2. Why do you start by trying to set some parameters and only then calling the begin method ... with much the same parameters?

Also also - this doesn't seem like an issue with the library, converting to discussion instead.

**Comment by Baptou88** [comment]
thank you for your reply
my programming knowledge is limited

I just deleted the line spi.begin and radio.begin fails

Before using your library I used that of sandeepmistry but when I switched to yours I foolishly copied my code without erasing the lines that had become useless

as for the power source, it is done by usb from my computer (I also tried with an external power supply without observing any change),
the antenna is the one that was sold with the board

**Reply by Baptou88** [reply]
I've noticed that on the Heltec V3 we have 2 connectors for antenna, I've plugged a second and the signal is at 56dbm!
[PXL_20221224_083800164.jpg](https://user-images.githubusercontent.com/48880112/209428153-8f99b190-7c5f-4f16-89cb-58cfe47b3d21.jpg)

**Reply by jgromes** [reply]
Then I would guess the second one is probably for the WiFi/BLE

**Reply by Baptou88** [reply]
thank you for the answer !
I find it a pity that the manufacturer does not provide this information...
